main.py

```diff
@@ -448,6 +470,278 @@ def move_methods_workaround(dex_filename, dex_filename_last, in_dir, out_dir, de
+    #
+    # Reassemble it
+    print_(" *** Reassembling classes...")
+    os.makedirs("out/")
+    #
+    try:
+        assemble(smali_folder, "out/"+dex_filename, DEVICE_SDK, True)
+        if sys.platform == "win32":
+            subprocess.check_call(["attrib", "-a", "out/"+dex_filename])
+    except subprocess.CalledProcessError as e:  # ToDO: Check e.cmd
+        safe_file_delete(TMP_DIR+"/out/"+dex_filename)  # Remove incomplete file
+        output = e.output.decode("utf-8")
+        if e.returncode != 1 or "Unsigned short value out of range: 65536" not in output:
+            print_(os.linesep+output.strip())
+            print_(os.linesep+"Return code: "+str(e.returncode))
+            exit_now(83)
+        warning("The reassembling has failed (probably we have exceeded the 64K methods limit)")
+        warning("but do NOT worry, we will retry.", False)
+        move_methods_workaround(dex_filename, dex_filename_last, "framework/", "out/", DEVICE_SDK)
+    #
+    # Backup the original file
+    BACKUP_FILE = os.path.join(OUTPUT_PATH, "framework.jar.backup")
+    safe_copy(os.path.join(TMP_DIR, "framework.jar"), BACKUP_FILE)
+    #
+    # Put classes back in the archive
+    print_(" *** Recompressing framework...")
+    compress(os.curdir+"/out/", "framework.jar")
+    #
+    # Copy the patched file to the output folder
+    print_(" *** Copying the patched file to the output folder...")
+    safe_copy(os.path.join(TMP_DIR, "framework.jar"), os.path.join(OUTPUT_PATH, "framework.jar"))
+    #
+    # Clean up "framework.jar" disassembly folder.
+    print_(" *** Deleting temporary directory:", TMP_DIR+"/smali-classes")
+    deltree(TMP_DIR+"/smali-classes")
+
+
+
+
+#
+# FUNCTION - main_patch_services_jar
+#
+def main_patch_services_jar():
+    print_(" *** Decompressing services.jar ...")
+    decompress("services.jar", "services/")
+    #
+    # Disassemble it
+    print_(" *** Disassembling services classes...")
+    smali_to_search = "com/android/server/pm/PackageManagerService.smali"
+    smali_folder, dex_filename, dex_filename_last = find_smali(smali_to_search, "services/", DEVICE_SDK)
+    #
+    # Check the existence of the file to patch
+    if smali_folder is None:
+        print_(os.linesep+"ERROR: The services.jar classes file to patch cannot be found, please report the problem.")
+        exit_now(82)
+    to_patch = smali_folder+smali_to_search
+    #
+    # Do the injection
+    print_(" *** Patching services...")
+    f = open(to_patch, "r")
+    old_contents = f.readlines()
+    f.close()
+    #
+    f = open(SCRIPT_DIR+"/patches/allowsdwrite.smali", "r")
+    allowsdwrite = f.readlines()
+    f.close()
+    #
+    # services_classes - Add allowsdwrite method
+    #       for testing - \smali-classes\com\android\server\pm
+    i = 0
+    contents = []
+    already_patched = False
+    in_function = False
+    required_line_before = False
+    right_line = False
+    start_of_line = None
```

Comment:
__[flake8]__
* [F841] local variable 'start_of_line' is assigned to but never used

<!-- Comment by [SideCI](https://sideci.com) -->